src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -346,7 +346,7 @@ public async ValueTask RefreshFileAsync(CancellationToken cancellationToken)
                     else
                     {
                         // The file isn't there anymore; do we still have the generator at all?
-                        if (project.AnalyzerReferences.Any(a => a.GetGenerators(project.Language).Any(static (g, self) => SourceGeneratorIdentity.GetGeneratorAssemblyName(g) == self._documentIdentity.Generator.AssemblyName, this)))
+                        if (project.AnalyzerReferences.Any(a => a.FullPath == _documentIdentity.Generator.AssemblyPath))
```

Comment:
is == ok on filepaths on windows?

Reply:
Generally it can be iffy, although in this case the AssemblyPath came from the analzyer reference's path in the first place. I'm going to leave this as is, since at some point this code needs to migrate to potentially run in VS for Mac too, where such a case insensitive comparison would actually be problematic, and no reason to add the extra overhead of the insensitive comparison if not needed.